### PR TITLE
Secondary IPD fixes

### PR DESCRIPTION
**Pull request by hyperlogic**
* vive puck calibration now works regardless of avatar scale
* Create mode will spawn entities with the correct orientation.
* Lasers on the 2d hud overlay now render correctly.

### CODE REVIEW

**Comment by hyperlogic** [comment]
Create Mode QA Test plan (no pucks or HMD required):
  * go to wikiplanet/newyork
  * enter create mode.
  * create one of each entity.
  * each entity should be aligned with your avatar's up orientation.  Not the (0, 1, 0) axis.

HUD Laser QA Test plan (HMD + hand controllers required)
  * Press Cnrl+b on the keyboard, a web browser entity should appear on the 2d HUD overlay.
  * You should be able to use the hand controller lasers to interact with the web browser.
  * Verify you can interact with the window around the web browser as well as the content within it.

Vive Puck Calibration (Vive HMD + Trackers required)
  * Try using all configurations of the Settings > Controller Settings.. calibration screen.
  * Including head and hand pucks.
  * your current avatar scale should not matter, and you should be able to scale your avatar after you calibrate with no problems.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3493/HighFidelity-Beta-PR11422-09c61deda835ccb8e66ccd779ade9be4a33b0dd6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3493/HighFidelity-Beta-PR11422-09c61deda835ccb8e66ccd779ade9be4a33b0dd6.exe

**Comment by YuppyPlays** [comment]
Testing: "HUD Laser QA Test plan (HMD + hand controllers required)"

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3498/HighFidelity-Beta-PR11422-5e93c13856ae0fa8eda25c68ffca55184bdfc305.exe

**Comment by YuppyPlays** [comment]
Testing: "HUD Laser QA Test plan (HMD + hand controllers required)"

**Comment by anshuman64** [comment]
build this please

**Comment by YuppyPlays** [comment]
Testing complete:

Press Cnrl+b on the keyboard, a web browser entity should appear on the 2d HUD overlay. **Check**
You should be able to use the hand controller lasers to interact with the web browser. **Check**
Verify you can interact with the window around the web browser as well as the content within it. **Check**

All looks good to me, the browser seems to work well, though the hands of the avatar end up behind it as if its overlaying them, though I assume this is intentional, if not though please do let me know. **Pass**

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3502/HighFidelity-Beta-PR11422-5e93c13856ae0fa8eda25c68ffca55184bdfc305.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3502/HighFidelity-Beta-PR11422-5e93c13856ae0fa8eda25c68ffca55184bdfc305.dmg

**Comment by cain-kilgore** [comment]
@hyperlogic: Can I get temp rez permissions granted on wikiplanet?

**Comment by anshuman64** [comment]
Vive Puck Calibration (Vive HMD + Trackers required)
- Feet = PASS
- Feet + Hips = PASS
- Feet + Hips + Chest = PASS
- Feet + Hips + Chest + Hands = PASS
- Feet + Hips + Chest + Hands + Head = FAIL. The calibration isn't working properly. Offset of one of the hand pucks is not being calculated correctly. Doesn't seem IPD related, just a bug, but should be fixed.
- Feet + Head = PASS. Leads me to believe the combination of Head & Hands (maybe chest too?) is the problem. 

Tried with Matthew, Matthew resized to 0.25, and Skeleton and tried resizing in between. Again, seems like you fixed calibration problems with IPD, seems like there is just a bug with head&hands calibration.

I can try combinations with shoulder pucks later today. And will try creating in Wikiplanet soon.

Nice work!

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/3548/HighFidelity-Beta-PR11422-54e627e2c59a6e25f8258e97ad4a00c52e3950fb.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/3548/HighFidelity-Beta-PR11422-54e627e2c59a6e25f8258e97ad4a00c52e3950fb.dmg

**Comment by anshuman64** [comment]
Did the entity spawning test. Looks good.

Updated the hand puck IK bug ticket: https://highfidelity.fogbugz.com/f/cases/7359/Hand-Puck-Arm-IK-Doesn-t-Work

Marking this PR as a QA Pass

File: plugins/openvr/src/ViveControllerManager.cpp [hunk 1]
```diff
@@ -676,40 +676,55 @@ void ViveControllerManager::InputDevice::uncalibrate() {
...
+        // use sensor space pose.
+        auto puckPoseIter = _validTrackedObjects.begin();
+        while (puckPoseIter != _validTrackedObjects.end()) {
+            if (puckPoseIter->first == puckIndex) {
+                break;
             }
+            puckPoseIter++;
+        }
+
+        //auto puckPoseIter = _poseStateMap.find(puckIndex);
```
Comment (sethalves) [inline plugins/openvr/src/ViveControllerManager.cpp]: leftover code

**Pull request merged**